easybuild/easyconfigs/f/FHI-aims/FHI-aims-240507-intel-2023a.eb

```diff
@@ -0,0 +1,63 @@
+##
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+#
+# Authors::   Dugan Witherick (University of Warwick)
+# Updated to 240507
+# Help from the FHI-aims developers and Alexander Grund to move to oneAPI is greatly appreciated
+# J. Sassmannshausen (Imperial College London/UK)
+##
+
+easyblock = 'CMakeMake'
+
+name = 'FHI-aims'
+version = '240507'
+
+homepage = 'https://fhi-aims.org/'
+description = """FHI-aims is an efficient, accurate all-electron,
+full-potential electronic structure code package for computational molecular  
+and materials science (non-periodic and periodic systems). The code supports  
+DFT (semilocal and hybrid) and many-body perturbation theory. FHI-aims is     
+particularly efficient for molecular systems and nanostructures, while        
+maintaining high numerical accuracy for all production tasks. Production      
+calculations handle up to several thousand atoms and can efficiently use (ten)thousands of cores.
+"""
+
+toolchain = {'name': 'intel', 'version': '2023a'}
+toolchainopts = {'opt': True, 'precise': True, 'usempi': True, 'oneapi': True}
+
+download_instructions = """
+The source code must be downloaded manually from the FHI-aims club
+(https://fhi-aims.org/get-the-code-menu/login).
+Access to the FHI-aims club requires a valid license and registration.
+Details on available license options and how to register to access
+FHI-aims club may be found at:
+https://fhi-aims.org/get-the-code-menu/get-the-code """
+
+sources = ['%(namelower)s.%(version)s.tar.gz']
+checksums = ['dea6483834d80ef5d8826171bd710ebcb035c4b971c4c958c82fdb9f03af6a6b']
+
+builddependencies = [('CMake', '3.26.3')]
+
+configopts = ' -DCMAKE_Fortran_COMPILER="$MPIF90" -DCMAKE_CXX_COMPILER="$MPICXX"  -DCMAKE_C_COMPILER="$MPICC" '
+configopts += ' -DLIBS="mkl_scalapack_lp64 mkl_blacs_intelmpi_lp64 mkl_intel_lp64 mkl_sequential mkl_core" '
+configopts += ' -DLIB_PATHS="$EBROOTIMKL/mkl/latest/lib/intel64" '
+configopts += ' -DCMAKE_Fortran_FLAGS="$FFLAGS -fc=ifx -finline-functions" '
+configopts += ' -DCMAKE_CXX_FLAGS="$CXXFLAGS -cxx=icpx -finline-functions -std=c++11" '
+configopts += ' -DCMAKE_C_FLAGS="$CFLAGS -cc=icx -finline-functions -std=gnu99" '
```

Comment:
Are you sure any of this is required?

- `-fc` etc. should be set by EasyBuild via the `I_MPI` variables
- finline-functions is set by `-O2` and above
- `-std` shouldn't be required as the defaults for those intel compilers are likely at least that

Removing might help updating the EC later.

Reply:
I took these flags from the template they provided. The `-finline-functions` comes from the dropping of `-ip` and that was the replacement I found. 
That said: you are right, it works without them as well. I will update the PR and will include one which makes use of an external `ELPA` installation.